site/app/libraries/GradingQueue.php

```diff
@@ -209,4 +211,220 @@ public function getGradingCount() {
+            if (in_array("required_capabilities", $entry->getQueueObj())) {
+                $capability = $entry->getQueueObj()["required_capabilities"];
+            }
+
+            if (!$is_grading) {
+                if (!array_key_exists($capability, $capability_queue_counts)) {
+                    $error .= "ERROR: {$job_file} requires {$capability} which is not provided by any worker";
+                }
+                else {
+                    $capability_queue_counts[$capability] += 1;
+
+                    $enabled_worker = false;
+                    foreach ($open_autograding_workers_json as $machine) {
+                        if ($machine["enabled"]) {
+                            if (in_array($capability, $machine["capabilities"])) {
+                                $enabled_worker = true;
+                            }
+                        }
+                    }
+
+                    if (!$enabled_worker) {
+                        $error .= "ERROR: {$job_file} requires {$capability} which is not provided by any *ENABLED* worker";
+                    }
+                }
+            }
+            else {
+                $max_time = $entry->getQueueObj()["max_possible_grading_time"];
+                $full_machine = $entry->getGradingQueueObj()['machine'];
+
+                $grading_machine = "NONE";
+                foreach ($open_autograding_workers_json as $machine => $details) {
+                    $m = $details["address"];
+                    if ($details["username"] != "") {
+                        $m = "{$details['username']}@{$m}";
+                    }
+                    if ($full_machine == $m) {
+                        $grading_machine = $machine;
+                        break;
+                    }
+                }
+                $machine_grading_counts[$grading_machine] += 1;
+
+                $elapsed_time = $entry->getElapsedTime();
+                if ($elapsed_time > $max_time) {
+                    $stale = true;
+                }
+            }
+            $file_segments = explode("__", $job_file);
+            if (!array_key_exists($file_segments[0] . "__" . $file_segments[1], $course_info)) {
+                $course_info[$file_segments[0] . "__" . $file_segments[1]] = [];
+            }
+
+            if (!array_key_exists($file_segments[2], $course_info[$file_segments[0] . "__" . $file_segments[1]])) {
+                $course_info[$file_segments[0] . "__" . $file_segments[1]][$file_segments[2]] = ["regrade" => 0, "interactive" => 0];
+            }
+
+            if ($regrade) {
+                $course_info[$file_segments[0] . "__" . $file_segments[1]][$file_segments[2]]["regrade"] += 1;
+            }
+            else {
+                $course_info[$file_segments[0] . "__" . $file_segments[1]][$file_segments[2]]["interactive"] += 1;
+            }
+        }
+        catch (\Exception $e) {
+            $error .= "Could not read for {$queue_or_grading_file}: {$e}";
```

Comment:
Do we want the entire stack trace here?

Reply:
An entire stack trace might be a bit excessive, I assume you are referring to the appending of `$e`?

Reply:
Yeah, I was unclear.  Currently it will print a large stack trace.  What I meant to suggest is that it be switched to `$e->getMessage()` to only print the message and not the entire stack trace.